core/src/main/scala/org/apache/spark/scheduler/TaskSetManager.scala

```diff
@@ -879,11 +889,12 @@ private[spark] class TaskSetManager(
       }
     }
 
-    if (successful(index)) {
-      logInfo(s"Task ${info.id} in stage ${taskSet.id} (TID $tid) failed, but the task will not" +
-        s" be re-executed (either because the task failed with a shuffle data fetch failure," +
-        s" so the previous stage needs to be re-run, or because a different copy of the task" +
-        s" has already succeeded).")
+    if (fetchFailedIndex.contains(index)) {
```

Comment:
This log info mainly inform the user that we won't re=execute the failed task, it makes little sense to differentiate between FetchFailed tasks and speculative tasks.